### PR TITLE
chore(flags): Add automatic cache warming on cache miss for /flags/definitions

### PR DESCRIPTION
**Pull request by haacked**
## Problem

The `/flags/definitions` endpoint is a Rust port of the `/local_evaluation` endpoint from Django. However, it does *not* implement building the response. Instead, it returns a HyperCached response built by the Django code.

We use Django signals to try and make sure we never have a cache miss, but any number of reasons could still lead to a cache miss such as pressure on Redis causing key evictions, etc.

## Changes

When the `/flags/definitions` endpoint encounters a cache miss, automatically trigger a cache rebuild via Redis queue. On the Django side, we have a celery-heartbeat running every second. When it sees the Redis queue entry, it rebuilds the cache.

**Cache rebuild mechanism:**
- Push notification to Redis list `posthog:flag_cache_miss_queue` with `team_id` and `timestamp`
- Celery task `process_flag_cache_miss_queue` polls this queue every second
- Rate limiting prevents excessive rebuilds (max 2 rebuilds per team per 60-second window by default)
- Cache rebuild is fire-and-forget - doesn't block the request response

**Response behavior:**
- **Total cache miss** (Redis + S3): Return HTTP 503 with `Retry-After: 2` header (configurable via
`FLAG_DEFINITIONS_CACHE_MISS_RETRY_AFTER_SECONDS`)
- **Partial cache miss** (Redis miss, S3 hit): Return HTTP 200 with S3 data, but still trigger cache warming to populate Redis
- **Redis cache hit**: Return HTTP 200, no cache warming needed

The 2-second interval was chosen because it covers the 1-second Celery beat interval + typical cache rebuild time (~276ms based on our testing). It's configurable via the `FLAG_DEFINITIONS_CACHE_MISS_RETRY_AFTER_SECONDS` setting.

## How did you test this code?

Unit Tests
Manually: (I tried all the cache scenarios: no cache, s3 only, fully cached).

## Changelog: (features only) Is this feature complete?

No

### CODE REVIEW

File: rust/feature-flags/src/cache/notification.rs [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+        .unwrap_or(0);
+
+    if queue_size >= *MAX_QUEUE_DEPTH {
+        warn!(
+            team_id = team_id,
+            queue_size = queue_size,
+            max_depth = *MAX_QUEUE_DEPTH,
+            "Cache miss queue overflow, dropping notification"
+        );
+        inc(FLAG_CACHE_MISS_NOTIFICATION_DROPPED_COUNTER, &[], 1);
```
[the remark this answers is not in the document]
Comment (haacked) [reply rust/feature-flags/src/cache/notification.rs:59]: Why the current approach is fine:

1. Soft limit by design - This is a fire-and-forget cache warming notification system. If we temporarily exceed 10,000 entries by a few dozen due to concurrent pushes, that's perfectly acceptable. We're not trying to enforce a hard limit for correctness.
2. Low likelihood of significant overflow - Even in a burst scenario, you'd need many concurrent cache misses at exactly the moment the queue is near capacity. Most of the time, the celery task is draining the queue every second.
3. Graceful degradation - If the queue does overflow temporarily, notifications get dropped and we have metrics tracking it. The system doesn't break - clients just retry in 2 seconds.
4. Complexity cost - Adding a Lua script increases cognitive load for anyone maintaining this code. You'd need to manage the script, handle its execution, and test edge cases.

**Comment by haacked** [comment]
To better understand how this is using Redis list and Celery Heartbeat, here's a description of the flow:

Step 1: Rust service pushes (LPUSH) to the LEFT:

```rust
// rust/feature-flags/src/cache/notification.rs:70
redis.lpush("posthog:flag_cache_miss_queue",
    r#"{"team_id": 123, "timestamp": 1699999999}"#
).await
```

Step 2: Celery task pops (RPOP) from the RIGHT:

```rust
# posthog/tasks/feature_flags.py:81
message = redis.rpop("posthog:flag_cache_miss_queue")
# Returns: '{"team_id": 123, "timestamp": 1699999999}'
``` 

Why `LPUSH + RPOP` = `FIFO` Queue
Redis List: "posthog:flag_cache_miss_queue"

```
Initial: []

After request 1 (LPUSH):
[{team: 1, ts: 100}]
 ↑ LEFT         RIGHT ↑

After request 2 (LPUSH):
[{team: 2, ts: 101}, {team: 1, ts: 100}]
 ↑ LEFT                          RIGHT ↑
   newest                        oldest

Celery task (RPOP):
[{team: 2, ts: 101}, {team: 1, ts: 100}]
                                    ↑
                              returns oldest first (FIFO)

After RPOP:
[{team: 2, ts: 101}]
 ↑ LEFT       RIGHT ↑
```

This creates a FIFO (First In, First Out) queue where:
- Producers (flags/definitions) add new items to the left
- Consumers (Celery) take old items from the right
- Oldest items get processed first.

File: posthog/tasks/feature_flags.py [hunk 2]
```diff
@@ -26,3 +31,100 @@ def sync_all_flags_cache() -> None:
...
+RATE_LIMIT_KEY_TEMPLATE = "posthog:flag_cache_rebuild_rate_limit:{team_id}"
+
+# Configurable operational parameters
+RATE_LIMIT_WINDOW = get_from_env("FLAG_CACHE_MISS_RATE_LIMIT_WINDOW", 60, type_cast=int)
+MAX_REBUILDS_PER_WINDOW = get_from_env("FLAG_CACHE_MISS_MAX_REBUILDS", 2, type_cast=int)
+BATCH_SIZE = get_from_env("FLAG_CACHE_MISS_BATCH_SIZE", 10, type_cast=int)
+QUEUE_DEPTH_WARNING_THRESHOLD = get_from_env("FLAG_CACHE_MISS_QUEUE_DEPTH_WARNING", 1000, type_cast=int)
+
+
+@shared_task(ignore_result=True, max_retries=0, queue=CeleryQueue.DEFAULT.value)
```
[the remark this answers is not in the document]
Comment (haacked) [reply posthog/tasks/feature_flags.py:55]: Hmm, I think this might be legit. We should have a dedicated Celery Queue for flags stuff.

[the remark this answers is not in the document]
Comment (haacked) [reply posthog/tasks/feature_flags.py:55]: Actually, we already are doing cache rebuilds on the DEFAULT queue. This poll task is very lightweight. I think we can make sure to measure and if there's issues, then we move it.
